### PR TITLE
Adding 'fixed_value' subdict and key to schema

### LINKED ISSUE #186: Support Fixed spans in Filters
## Type of Issue

Select the type of issue:
- [ ] Bug report (to report a bug)
- [x] Feature request (to request an additional feature)
- [ ] Tracker (I am just using this as a tracker)
- [ ] Refactor request
- [ ] Documentation Ask

## Description

Add a key for fixed spans in datasets, and add a new head in parser to predict this.

## Current Behavior

Examples of fixed spans:
```
{"reference_object": {"special_reference": "SPEAKER"}}
```
```
{"pred_text": "has_tag", "obj_text": "_currently_running"}
```

## Expected Behavior

```
{
  "pred_text": "has_tag", 
  "obj_text": {"fixed_value": "SPEAKER"}
}
{"pred_text": "has_tag", 
 "obj_text": {"fixed_value": "CURRENTLY_RUNNING"}
}
```

## Links to any relevant pastes or documents 

https://docs.google.com/document/d/1AlXJTUxQmnuzPZSsWa0I0X90tkfmGDLjVo261zeJdwc/

## Checklist

Use this checklist if you are using this issue as a tracker:

- [ ] Support fixed spans in datasets
    - [ ] annotated
    - [ ] templated
    - [ ] templated filters
- [ ] Add head to transformer model
    - [ ] softmax over fixed set
    - [ ] ranking loss over all in memory
- [ ] Support fixed spans in interpreter
- [ ] Update grammar spec
    - [ ] Markdown
    - [ ] JSON validator
    - [ ] Autocomplete tool

### PR DESCRIPTION
**Pull request by kavyasrinet**
# Description

This PR proposes a schema change to support special 'fixed_value' for some keys.

Proposal to fix #186 

## Type of change

Please check the options that are relevant.

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] Proposes a change (non-breaking change that isn't necessarily a bug)
- [ ] Refactor
- [ ] New feature (non-breaking change that adds a new functionality)
- [ ] Breaking change (fix or feature that would break some existing functionality downstream)
- [ ] This is a unit test
- [x] Documentation only change

## Type of requested review

- [x] I want a thorough review of the implementation.
- [ ] I want a high level review. 
- [ ] I want a deep design review.

### CODE REVIEW

**Review by Rebecca-Qian — COMMENTED** [review]
Can we add a set of literals that can be fixed values for now? As a placeholder until we have a better canonicalizer.

cc @aszlam

File: base_agent/documents/logical_form_specification/filters.md [hunk 1]
```diff
@@ -55,8 +55,8 @@ This is used to mean the number of steps (in "has_measure" units, default is "bl
 {"linear_extent" : {
             "relative_direction": "LEFT" / "RIGHT"/ "UP" / "DOWN"/ "FRONT" 
                                   / "BACK"/ "AWAY" / "INSIDE" / "OUTSIDE", 
-            "frame": "SPEAKER" / "AGENT" / "ABSOLUTE" / {"player_span": span},
-            "has_measure" : span,
+            "frame": {"fixed_value": "SPEAKER" / "AGENT" / "ABSOLUTE", "player_span": span},
```
Comment (aszlam) [inline base_agent/documents/logical_form_specification/filters.md:58]: slightly not clear from this, bc the comma: it should either be fixed_value xor player_span, but we have been using the comma to denote possibility of non-exclusive or

File: base_agent/documents/logical_form_specification/human_give_command.md [hunk 2]
```diff
@@ -287,9 +287,9 @@ where `has_x` can be : `has_block_type`, `has_name`, `has_size`, `has_orientatio
   "yaw_pitch": span,
   "yaw": span,
   "pitch": span,
-  "relative_yaw" = {"angle": -360, -180, -135, -90, -45, 45, 90, 135, 180, 360 
+  "relative_yaw" = {"fixed_value": -360 / -180 / -135 / -90 / -45 / 45 / 90 / 135 / 180 / 360,
```
Comment (aszlam) [inline base_agent/documents/logical_form_specification/human_give_command.md:290]: same comment above- I assume this means xor, but comma is used differently than previous

**Review by aszlam — APPROVED** [review]

File: base_agent/documents/logical_form_specification/filters.md (hunk 1 above)
Comment (kavyasrinet) [reply base_agent/documents/logical_form_specification/filters.md:58]: Right yeah: so we'd either have "fixed_value" which can have one of the values (the list of which is separated by `/`) or we have "player_span". 
Would something like this make it clearer:
```
"frame": {"fixed_value": "SPEAKER" / "AGENT" / "ABSOLUTE"}  /  {"player_span": span}
```

File: base_agent/documents/logical_form_specification/human_give_command.md (hunk 2 above)
Comment (kavyasrinet) [reply base_agent/documents/logical_form_specification/human_give_command.md:290]: Same reply as above.

**Pull request merged**